src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/StandaloneMapMarkersPlugin.java

```diff
@@ -112,94 +104,55 @@ void addRow(double lat, double lon, String overlayValue) {
-    boolean anyHaveBinStart = binRanges.stream().filter(bin -> bin.getBinStart() != null).findFirst().isPresent();
-    boolean anyHaveBinEnd = binRanges.stream().filter(bin -> bin.getBinEnd() != null).findFirst().isPresent();
-    boolean allHaveBinStart = !binRanges.stream().filter(bin -> bin.getBinStart() == null).findFirst().isPresent();
-    boolean allHaveBinEnd = !binRanges.stream().filter(bin -> bin.getBinEnd() == null).findFirst().isPresent();
-    if ((anyHaveBinStart | anyHaveBinEnd) & !(allHaveBinStart & allHaveBinEnd)) {
-      throw new ValidationException("All BinRanges must have binStart and binEnd if any BinRange has either.");
+  protected void validateOverlayConfig(OverlayConfig overlayConfig) throws ValidationException {
+    switch (overlayConfig.getOverlayType()) {
+      case CONTINOUS -> validateContinousBinRanges((ContinousOverlayConfig) overlayConfig);
+      case CATEGORICAL -> validateCategoricalOverlayValues((CategoricalOverlayConfig) overlayConfig);
     }
+  }
 
-    // start is always before end
-    boolean anyStartAfterEnd = binRanges.stream().filter(bin -> bin.getBinEnd() <= bin.getBinStart()).findFirst().isPresent();
-    if (anyHaveBinStart & anyStartAfterEnd) {
-      throw new ValidationException("All BinRanges must have binStart less than binEnd.");
+  private void validateCategoricalOverlayValues(CategoricalOverlayConfig overlayConfig) throws ValidationException {
+    if (!getUtil().getVariableDataShape(overlayConfig.getOverlayVariable()).equalsIgnoreCase(APIVariableDataShape.CATEGORICAL.toString())) {
+      throw new ValidationException("Input overlay variable %s is %s, but provided overlay configuration is for a categorical variable"
+          .formatted(overlayConfig.getOverlayVariable().getVariableId(), getUtil().getVariableDataShape(overlayConfig.getOverlayVariable())));
     }
-
-    // no duplicate labels
-    List<String> labels = new ArrayList<String>();
-    binRanges.forEach( bin -> labels.add(bin.getBinLabel()) );
-    if (hasDuplicates(labels)) {
-      throw new ValidationException("All BinRanges must have unique binLabels.");
-    }    
-
-    // ranges must not overlap
-    // TODO again need to add support for dates here
-    if (anyHaveBinStart) {
-      Map<Double, Double> binEdges = new TreeMap<Double, Double>();
-      binRanges.forEach( bin -> binEdges.put(Double.parseDouble(bin.getBinStart()), Double.parseDouble(bin.getBinEnd())));
-      boolean first = true;
-      for (Double binStart : binEdges.keySet()) {
-        Double prevBinEnd;
-        if (first) {
-          prevBinEnd = binEdges.get(binStart);
-          first = false;
-        } else {
-          if (prevBinEnd > binStart) {
-            throw new ValidationException("Bin Ranges must not overlap");
-          }
-          prevBinEnd = binEdges.get(binStart);
-        }
-      }
+    int numDistinctOverlayVals = new HashSet<>(overlayConfig.getOverlayValues()).size();
```

Comment:
i feel like some of this validation isnt unique to map markers, but just the overlay generally, which will be used for all floating charts as well.

Reply:
Ah I see, I'm a little behind on the requirements for the floating plugins. Are there existing mock-ups that you know of?

Reply:
Hahahahaha... No.